components/experimental/src/units/measureunit.rs

```diff
@@ -102,48 +115,39 @@ impl<'data> MeasureUnitParser<'data> {
                         let (si_prefix, identifier_part_without_si_prefix) =
                             self.get_si_prefix(identifier_part_without_power);
                         let (unit_id, identifier_part_without_unit_id) =
-                            self.get_unit_id(identifier_part_without_si_prefix)?;
+                            match self.get_unit_id(identifier_part_without_si_prefix) {
+                                Ok((unit_id, identifier_part_without_unit_id)) => {
+                                    (unit_id, identifier_part_without_unit_id)
+                                }
+                                Err(_) if identifier.starts_with(b"per-") => {
+                                    if sign < 1 {
+                                        return Err(ConversionError::InvalidUnit);
+                                    }
+                                    sign = -1;
+                                    identifier = &identifier[4..];
+                                    continue;
+                                }
```

Comment:
We don't allow `[]` in non-experimental code because it can panic. It would be better to use [strip_prefix](https://doc.rust-lang.org/std/primitive.slice.html#method.strip_prefix).

Untested, but something like this:

```suggestion
                                Err(_) if let Some(remainder) = identifier.strip_prefix(b"per-") => {
                                    if sign < 1 {
                                        return Err(ConversionError::InvalidUnit);
                                    }
                                    sign = -1;
                                    identifier = remainder;
                                    continue;
                                }
```